src/com/sun/javatest/Script.java

```diff
@@ -477,6 +484,49 @@ public void run() {
         }
     }
 
+
+    /**
+     * This method runs the statusTransformer stored in the private variable. In case the statusTransformer is null
+     * it returns the original status.
+     * @param originalStatus - original status of the testrun
+     * @param td - description of the test
+     * @return status after modification (the originalStatus by default)
+     */
+    private Status tryModifyStatus(Status originalStatus, TestDescription td){
+        if (statusTransformer == null) {
+            return originalStatus;
+        }
+        //in case the status has been modified by this function we want to log it for any user controlling the results.
+        Status newStatus = statusTransformer.transform(originalStatus, td);
+        newStatus = new Status(newStatus.getType(), newStatus.getReason() +
+                    " - The status of this test result has been modified by external code.");
```

Comment:
What do you think of mentioning the transformer name?

```suggestion
                    " - The status of this test result has been modified by " + statusTransfomer.getClass().getName());
```

Reply:
might be nice, will add that. Thanks for your suggestion :-)